daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -612,44 +612,38 @@ private[lf] object SBuiltin {
     }
   }
 
-  final case object SBGenMapToList extends SBuiltinPure(1) {
+  final case object SBMapToList extends SBuiltinPure(1) {
 
     override private[speedy] final def executePure(args: util.ArrayList[SValue]): SValue =
       SValue.toList(getSMap(args, 0).entries)
   }
 
-  final case object SBGenMapInsert extends SBuiltinPure(3) {
-    override private[speedy] final def executePure(args: util.ArrayList[SValue]): SValue = {
-      val genMap = getSMap(args, 2)
-      val key = getSMapKey(args, 0)
-      SGenMap(genMap.isTextMap, genMap.entries.updated(key, args.get(1)))
-    }
+  final case object SBMapInsert extends SBuiltinPure(3) {
+    override private[speedy] final def executePure(args: util.ArrayList[SValue]): SValue =
+      getSMap(args, 2).update(getSMapKey(args, 0), args.get(1))
```

Comment:
Complexity was wrong here are the map was completely resorted by the constructor.